widgets/gui/qt_main_window.py

```diff
@@ -11,15 +15,59 @@ def __init__(self, *args, **kwargs):
         self.title = "Pisces Parameter Controller"
         self.left = 10
         self.top = 10
-        self.width = 500
+        self.width = 700
         self.height = 500
 
         self.setWindowTitle(self.title)
         self.setGeometry(self.left, self.top, self.width, self.height)
+        self.setDockOptions(QMainWindow.AnimatedDocks | QMainWindow.AllowNestedDocks)
+
+        # initialize docks
+        self.live_dock = QDockWidget("Live", self)
+        self.timelapse_dock = QDockWidget("Timelapse", self)
+        self.parameters_dock = QDockWidget("Parameters", self)
+
+        # set common configurations for docks
+        self.dock_list = [self.live_dock, self.timelapse_dock, self.parameters_dock]
+
+        for dock in self.dock_list:
+            dock.setFeatures(
+                QDockWidget.DockWidgetClosable
+                | QDockWidget.DockWidgetMovable
+                | QDockWidget.DockWidgetFloatable
+            )
+            dock.setAllowedAreas(Qt.LeftDockWidgetArea | Qt.RightDockWidgetArea | Qt.TopDockWidgetArea)
+
+        # set maximum dock sizes
+        self.live_dock.setMaximumSize(200, 140)
+        self.timelapse_dock.setMaximumSize(200, 140)
+        self.parameters_dock.setMaximumSize(500, 650)
+
+        # initialize widgets and assign to their dock
+        self.live_widget = LiveControl(self, "Live Mode")
+        self.live_dock.setWidget(self.live_widget)
+        self.addDockWidget(Qt.RightDockWidgetArea, self.live_dock)
+
+        self.timelapse_widget = TimelapseControl(self, "Timelapse Mode")
+        self.timelapse_dock.setWidget(self.timelapse_widget)
+        self.addDockWidget(Qt.RightDockWidgetArea, self.timelapse_dock)
+
+        self.parameters_widget = ParametersWidget(self)
+        self.parameters_dock.setWidget(self.parameters_widget)
+        self.addDockWidget(Qt.LeftDockWidgetArea, self.parameters_dock)
+
+        # split horizontal and vertical space between docks
+        self.splitDockWidget(self.parameters_dock, self.live_dock, Qt.Horizontal)
```

Comment:
Also, let me know if this is a good way to split. I wasn't sure how to split it given live/timelapse take up the same horizontal space initially.